src/main/java/com/powsybl/openloadflow/dc/DcLoadFlowEngine.java

```diff
@@ -205,8 +208,41 @@ public DcLoadFlowResult run() {
         double initialSlackBusActivePowerMismatch = getActivePowerMismatch(network.getBuses());
         double distributedActivePower = 0.0;
         if (parameters.isDistributedSlack() || parameters.isAreaInterchangeControl()) {
-            // FIXME handle distribution failure
-            distributedActivePower = distributeSlack(network, network.getBuses(), parameters.getBalanceType(), parameters.getNetworkParameters().isUseActiveLimits());
+            LoadFlowParameters.BalanceType balanceType = parameters.getBalanceType();
+            boolean useActiveLimits = parameters.getNetworkParameters().isUseActiveLimits();
+            ActivePowerDistribution activePowerDistribution = ActivePowerDistribution.create(balanceType, false, useActiveLimits);
```

Comment:
I inlined it to avoid a double-call to getActivePowerMismatch, indeed we should perhaps move distributeSlack() to Woodbury, and create an issue about usage of slack distribution failure behavior in Woodbury engine